projects/vdk-core/src/vdk/internal/builtin_plugins/run/noop_datajob.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pathlib
+from pathlib import Path
+from typing import cast
+
+from vdk.api.job_input import IJobInput
+from vdk.api.plugin.core_hook_spec import CoreHookSpecs
+from vdk.api.plugin.core_hook_spec import JobRunHookSpecs
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.internal.builtin_plugins import builtin_hook_impl
+from vdk.internal.builtin_plugins.internal_hookspecs import InternalHookSpecs
+from vdk.internal.builtin_plugins.run.data_job import DataJobDefaultHookImplPlugin
+from vdk.internal.builtin_plugins.run.data_job import JobArguments
+from vdk.internal.builtin_plugins.run.file_based_step import TYPE_PYTHON
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.builtin_plugins.run.step import Step
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.internal.core.context import CoreContext
+from vdk.internal.core.statestore import StateStore
+from vdk.internal.plugin.plugin import PluginRegistry
+
+
+class NoOpStepDataJobHookImplPlugin(DataJobDefaultHookImplPlugin):
+    """
+    Implementation that overrides of the functionality of the DataJobDefaultHookImplPlugin
+    so that it runs a single NoOp step
+    """
+
+    @hookimpl
+    def initialize_job(self, context: JobContext):
+        step = Step(
+            name=f"NoOpStep",
+            type=TYPE_PYTHON,
+            runner_func=lambda _step, _job_input: True,  # Intentional noop
+            file_path=Path(__file__),
+            job_dir=context.job_directory,
+        )
+        context.step_builder.add_step(step)
+
+
+class NoOpStepDataJob:
+    """
+    DataJob that exposes methods to enable running each phase of the DataJob lifecycle
+    independently and get access to an initialised IJobInput object after the run_and_return_job_input() step
+
+    Sample usage:
+            datajob_directory = Path(abspath(file_relative_path(__file__, '../')))
+            datajob = NoOpStepDataJob(datajob_directory)
+            try:
+                datajob.initialize_job()
+                vdk_job_input = datajob.run_and_return_job_input()
+
+                ... use vdk_job_input object to interact with SuperCollider
+            finally:
+                datajob.finalize_job()
```

Comment:
I am not sure about NoOpStepDataJob name but I can't think of a better one.  

I would prefer to hide initailzie_job and finalize_job because 
A) I'd like to minimize the new interfaces we expose . 
B) Make it simpler to use (without requiring to call manually methods)
C) The implementation of Data Job run(execution) cycle algorithm [here](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py#L255) is something that in the future may change and even become a plugin hook itself (so customizable) so it's better it's a bit hidden.

In python often context manager may be used to achieve that (https://docs.python.org/3/library/contextlib.html )

So maybe something like 

```
class NoOpDataJob: 
   def __enter__(self)
     # effectively this is hidden from user and we can change it if 
      self.job.initialize_job 

  def __exit__(self): 
     self.job.finalize_job()
```

and would be used like 
```
with NoOpDataJob() as job:
    job.get_job_input() 
```

Reply:
@tozka I took a stab at hiding the `initialize_job()` and `finalize_job()` in f01eaf6 using a contextmanager as suggested, enabling usage like this:

```
with InitialisedJobInput(datajob_directory) as vdk_job_input:
    #... use vdk_job_input object to interact with SuperCollider
```

Notes:
1.  I really like how this cleans up usage & reduces the API surface area
2. Renamed `NoOpDataJob` to `InitialisedJobInput` - still not sure about this name

WDYT?

Reply:
I didn't like the "NoOp" part because using plugins you can provide actual steps to the job so it might not be NoOpStep. You simply add a noOp step to avoid the validation for empty steps but that's an implementation detail. 


I still think it's a DataJob though not JobInput. We simply expose job input for further usage. Maybe InitializedDataJob or StandaloneDataJob. Something like that.

Here we are inverting the flow of control really. From more declarative (where the job is executed by framework (vdk run)) to more imperative - the user decides when the job would be executed.